workspace_tools/toolchains/gcc.py

```diff
@@ -69,11 +69,8 @@ def __init__(self, target, options=None, notify=None, macros=None, silent=False,
         if "save-asm" in self.options:
             common_flags.append("-save-temps")
 
-        if "debug-info" in self.options:
-            common_flags.append("-g")
-            common_flags.append("-O0")
-        else:
-            common_flags.append("-O2")
+        common_flags.append("-ggdb")
```

Comment:
Yes, that was for debug purposes. Can be reverted to original version.

From: Martin Kojtal [mailto:notifications@github.com]
Sent: 24 września 2015 16:01
To: mbedmicro/mbed
Cc: Górniak Wojciech
Subject: Re: [mbed] added Espotel Elmo platform (#1341)


In workspace_tools/toolchains/gcc.py<https://github.com/mbedmicro/mbed/pull/1341#discussion_r40321151>:

> @@ -69,11 +69,8 @@ def __init__(self, target, options=None, notify=None, macros=None, silent=False,

>          if "save-asm" in self.options:

>              common_flags.append("-save-temps")

>

> -        if "debug-info" in self.options:

> -            common_flags.append("-g")

> -            common_flags.append("-O0")

> -        else:

> -            common_flags.append("-O2")

> +        common_flags.append("-ggdb")

Why did you remove debug-info? Means to compile always with opt 0?

—
Reply to this email directly or view it on GitHub<https://github.com/mbedmicro/mbed/pull/1341/files#r40321151>.